pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -243,6 +239,36 @@ func (p *PoolManager) initVirtualMachineMap() error {
 		return nil
 	}
 
+	err := p.forEachVmInterfaceInClusterRunFunction(func(vmFullName string, iface kubevirt.Interface, networks map[string]kubevirt.Network) error {
+		if !validateInterfaceSupported(iface, networks) {
+			return nil
+		}
+
+		if iface.MacAddress != "" {
+			if err := p.allocateRequestedVirtualMachineInterfaceMac(vmFullName, iface, logger); err != nil {
+				// Dont return an error here if we can't allocate a mac for a configured vm
+				logger.Error(err, "Invalid/Duplicate mac address for virtual machine",
+					"virtualMachineFullName", vmFullName,
+					"virtualMachineInterfaceMac", iface.MacAddress)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to iterate the cluster vm interfaces to recreate the macPoolMap")
+	}
+
+	err = p.initMacMapFromLegacyConfigMap()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// forEachMacInClusterRunFunction gets all the macs from all the supported interfaces in all the cluster vms, and runs
+// a function f on it
+func (p *PoolManager) forEachVmInterfaceInClusterRunFunction(f func(vmFullName string, iface kubevirt.Interface, networks map[string]kubevirt.Network) error) error {
```

Comment:
Can we use getvmInstance here so we reuse the RESTClient call ?

Reply:
it's a different request, here we get all the vms in a list, and in `getvmInstance` we only get one vm.. so the URIs and target struct are different